### DISCUSSION TITLE
Open source Counterparty data explorer

### DISCUSSION THREAD

**Discussion by jotapea** [CounterpartyXCP/CIPs#115] [General]
An open source Counterparty data explorer is live at [xcp.dev](https://xcp.dev)!

Some notable features:
- Navigate all transactions: https://www.xcp.dev/transactions
- Navigate all protocol messages: https://www.xcp.dev/messages
- Address view (with all its dispensers, broadcasts, and more in a single page): https://www.xcp.dev/address/1Pcpxw6wJwXABhjCspe3CNf3gqSeh6eien
- Asset view (with all its issuance history, market availability, and more in a single page): https://www.xcp.dev/asset/TEST

The explorer also includes [wallet tools](https://www.xcp.dev/wallet), which allow users to see balances and generate (unsigned) transactions without private keys (the project also provides a [simple wallet](https://github.com/CNTRPRTY/simplest) that can be used for signing these transactions).

The [repository](https://github.com/CNTRPRTY/xcpdev) includes everything needed (full-stack):
- [Counterparty node setup](https://github.com/CNTRPRTY/xcpdev/tree/main/server/fednode)
- [Basic Node.js based backend code](https://github.com/CNTRPRTY/xcpdev/tree/main/server/express_server)
- [React based web frontend](https://github.com/CNTRPRTY/xcpdev/tree/main)

Feel free to leave any feedback in this discussion, its repo, or in [t.me/xcpdev](https://t.me/xcpdev).

**Comment by adamkrellenstein** [comment]
This explorer is currently running a fork of Counterparty without advertising it as such and without even including replay protection.
